app/src/lib/stores/app-store.ts

```diff
@@ -1451,6 +1451,13 @@ export class AppStore extends TypedBaseStore<IAppState> {
     previousRepositoryId: number | null,
     currentRepositoryId: number
   ) {
+    // No need to update the recent repositories if the selected repository is
+    // the same as the old one (this could happen when the alias of the selected
+    // repository is changed).
+    if (previousRepositoryId === currentRepositoryId) {
+      return
+    }
```

Comment:
Without this, creating/changing/removing the alias of the current repo, would move the current repo into the Recents list, which doesn't make sense.